packages/netstring/reader.js

```diff
@@ -67,6 +67,8 @@ async function* makeNetstringIterator(
       `Unexpected dangling message at offset ${offset} of ${name}`,
     );
   }
+
+  return undefined;
```

Comment:
I’m not sure about prevention, but I have encountered difficulty and backed off to `undefined` by default for all the templates of a stream. The problem arises from the symmetry of the type. Void is great for a type that will be returned and can be ignored, where unknown seems to work better for an argument, but with pipe and pump, args become returns and returns become args. But, this is something I want to improve upon when I understand the base principles and emergent effects better. For example, it would be nice to have overloads that allow an argument of type undefined to be omitted more generally.